addons/stock_account/stock.py

```diff
@@ -260,8 +260,12 @@ def _get_invoice_vals(self, cr, uid, key, inv_type, journal_id, move, context=No
         else:
             account_id = partner.property_account_payable.id
             payment_term = partner.property_supplier_payment_term.id or False
+        if move.picking_id.name is move.origin:
```

Comment:
I'm not sure this test is useful, because I think we're always coming from a Delivery order in this method.